app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,29 @@
+class SessionsController < ApplicationController
+  def front
+    redirect_to videos_path if sign_in?
+  end
+
+  def new
+    redirect_to videos_path if sign_in?
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
```

Comment:
Yes, `find_by_email` does not exist, so it must be dynamically created. This is what makes it inefficient compared to `find_by`, which was introduced in Rails 4. Just remember, when you search for 1 record use `find_by`(only) and for multiple records use `where`.

Also check this Railscast - http://railscasts.com/episodes/400-what-s-new-in-rails-4?autoplay=true 

from 6:10